desktop/portal/launcher.go

```diff
@@ -0,0 +1,192 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020-2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package portal
+
+import (
+	"fmt"
+	"syscall"
+	"time"
+
+	"github.com/godbus/dbus"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+const (
+	desktopPortalBusName      = "org.freedesktop.portal.Desktop"
+	desktopPortalObjectPath   = "/org/freedesktop/portal/desktop"
+	desktopPortalOpenURIIface = "org.freedesktop.portal.OpenURI"
+	desktopPortalRequestIface = "org.freedesktop.portal.Request"
+)
+
+type ResponseError struct {
+	msg string
+}
+
+func (u *ResponseError) Error() string { return u.msg }
+
+func (u *ResponseError) Is(err error) bool {
+	_, ok := err.(*ResponseError)
+	return ok
+}
+
+func MakeResponseError(msg string) error {
+	osutil.MustBeTestBinary("only to be used in tests")
+	return &ResponseError{msg: msg}
+}
+
+// Launcher is a launcher that forwards the requests to xdg-desktop-portal DBus API
+type Launcher struct{}
```

Comment:
As mentioned in previous comments, I don't think there is much benefit in grouping the code below as methods on a type.  We're not trying to implement xdgopenproxy's desktopLauncher interface any more, so just use functions.

Reply:
OK. I was actually thinking of moving the D-Bus connection to the constructor of the Launcher type, but this can also wait -- unless we plan to add more methods to the interface, I guess that removing the type is fine.

Done.